src/Uno.UI/UI/Xaml/Controls/Primitives/RepeatButton/RepeatButton.mux.cs

```diff
@@ -0,0 +1,377 @@
+			{
+				_pointerCausingRepeat = true;
+			}
+
+			UpdateRepeatState();
+			UpdateVisualState();
+		}
+
+		/// <summary>
+		/// PointerMoved event handler.
+		/// </summary>
+		/// <param name="args">Event args.</param>
+		protected override void OnPointerMoved(PointerRoutedEventArgs args)
+		{
+			// The reason this function does not call the base 
+			// OnPointerMove is that the ButtonBase class sometimes 
+			// sets IsPressed to false based on mouse position. This 
+			// interferes with the RepeatButton functionality, which
+			// relies on IsPressed being True in Hover mode incorrectly.						
+		}
+
+		/// <summary>
+		/// PointerLeave event handler.
+		/// </summary>
+		/// <param name="args">Event args.</param>
+		protected override void OnPointerExited(PointerRoutedEventArgs args)
+		{
+			if (ShouldIgnoreInput(args))
+			{
+				return;
+			}
+
+			base.OnPointerExited(args);
+
+			if (ClickMode == ClickMode.Hover)
+			{
+				_pointerCausingRepeat = false;
+				UpdateRepeatState();
+			}
+
+			UpdateVisualState();
+		}
+
+		/// <summary>
+		/// PointerPressed event handler.
+		/// </summary>
+		/// <param name="args">Event args.</param>
+		protected override void OnPointerPressed(PointerRoutedEventArgs args)
+		{
+			if (args.Handled)
+			{
+				return;
+			}
+
+			if (ShouldIgnoreInput(args))
+			{
+				return;
+			}
+
+			base.OnPointerPressed(args);
+
+			var pointerPoint = args.GetCurrentPoint(this);
+			var isLeftButtonPressed = pointerPoint.Properties.IsLeftButtonPressed;
+
+			if (isLeftButtonPressed)
+			{
+				if (ClickMode != ClickMode.Hover)
+				{
+					_pointerCausingRepeat = true;
+					UpdateRepeatState();
+				}
+			}
+		}
+
+		/// <summary>
+		/// PointerReleased event handler.
+		/// </summary>
+		/// <param name="args">Event agrs.</param>
+		protected override void OnPointerReleased(PointerRoutedEventArgs args)
+		{
+			if (args.Handled)
+			{
+				return;
+			}
+
+			if (ShouldIgnoreInput(args))
+			{
+				return;
+			}
+
+			base.OnPointerReleased(args);
+
+			if (ClickMode != ClickMode.Hover)
+			{
+				_pointerCausingRepeat = false;
+				UpdateRepeatState();
+			}
+
+			UpdateVisualState();
+		}
+
+		protected override AutomationPeer OnCreateAutomationPeer() => new RepeatButtonAutomationPeer(this);
+
+		private void StartTimer()
+		{
+			if (_timer == null)
+			{
+				_timer = new DispatcherTimer();
+				_timer.Tick += TimerCallback;
+			}
+
+			if (!_timer.IsEnabled)
+			{
+				_timer.Interval = TimeSpan.FromMilliseconds(Delay);
+				_timer.Start();
+			}
+		}
+
+		private void StopTimer() => _timer?.Stop();
+
+		private void UpdateRepeatState()
+		{
+			if ((_pointerCausingRepeat && IsPointerOver) || _keyboardCausingRepeat)
+			{
+				StartTimer();
+			}
+			else
+			{
+				StopTimer();
+			}
+		}
+
+		private void TimerCallback(object? sender, object state)
+		{
+			var interval = TimeSpan.FromMilliseconds(Interval);
+
+			if (_timer!.Interval != interval)
+			{
+				_timer.Interval = interval;
+			}
```

Comment:
Is this coming from WinUI? Otherwise why are you not just changing the `_timer.Interval` in the `OnIntervalChanged` callback? Changing the `Interval` while the timer is running is expected to [restart the timer](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/DispatcherTimer.cs#L44) and apply the updated interval immediately (the `Tick` will be raised immediately if the new interval is below the elapsed time since last tick).

Reply:
The code comes from WinUI, so I guess they expect the next tick to occur with the old interval and only then get updated, although you are definitely right that your approach would make much more sense.